aws-ssm-document/src/main/java/com/amazonaws/ssm/document/CreateHandler.java

```diff
@@ -74,29 +79,48 @@ public CreateHandler() {
             createDocumentRequest =
                     documentModelTranslator.generateCreateDocumentRequest(model, request.getSystemTags(),
                         request.getDesiredResourceTags(), request.getClientRequestToken());
+
         } catch (final InvalidDocumentContentException e) {
             throw new CfnInvalidRequestException(e.getMessage(), e);
         }
 
         model.setName(createDocumentRequest.name());
 
         try {
-            final CreateDocumentResponse response = proxy.injectCredentialsAndInvokeV2(createDocumentRequest, ssmClient::createDocument);
+            final CreateDocumentResponse response = createDocument(createDocumentRequest, model, proxy, logger);
+
             context.setCreateDocumentStarted(true);
             context.setStabilizationRetriesRemaining(NUMBER_OF_DOCUMENT_CREATE_POLL_RETRIES);
 
             return ProgressEvent.<ResourceModel, CallbackContext>builder()
-                    .resourceModel(model)
-                    .status(OperationStatus.IN_PROGRESS)
-                    .message(response.documentDescription().statusInformation())
-                    .callbackContext(context)
-                    .callbackDelaySeconds(CALLBACK_DELAY_SECONDS)
-                    .build();
+                .resourceModel(model)
+                .status(OperationStatus.IN_PROGRESS)
+                .message(response.documentDescription().statusInformation())
+                .callbackContext(context)
+                .callbackDelaySeconds(CALLBACK_DELAY_SECONDS)
+                .build();
         } catch (final SsmException e) {
             throw exceptionTranslator.getCfnException(e, model.getName(), OPERATION_NAME, logger);
         }
     }
 
+    private CreateDocumentResponse createDocument(final CreateDocumentRequest createDocumentRequest,
+                                                  final ResourceModel model,
+                                                  final AmazonWebServicesClientProxy proxy,
+                                                  final Logger logger) {
+
+        try {
+            return proxy.injectCredentialsAndInvokeV2(createDocumentRequest, ssmClient::createDocument);
```

Comment:
does it require additional permissions if it goes through create api call?

Reply:
No additional permissions are required here